libs/vkd3d/resource.c

```diff
@@ -1727,6 +1734,98 @@ HRESULT vkd3d_sampler_state_create_static_sampler(struct vkd3d_sampler_state *st
     return S_OK;
 }
 
+static VkResult vkd3d_sampler_state_create_descriptor_pool(struct d3d12_device *device, VkDescriptorPool *vk_pool)
+{
+    const struct vkd3d_vk_device_procs *vk_procs = &device->vk_procs;
+    VkDescriptorPoolCreateInfo pool_info;
+    VkDescriptorPoolSize pool_size;
+
+    pool_size.type = VK_DESCRIPTOR_TYPE_SAMPLER;
+    pool_size.descriptorCount = 16384;
+
+    pool_info.sType = VK_STRUCTURE_TYPE_DESCRIPTOR_POOL_CREATE_INFO;
+    pool_info.pNext = NULL;
+    pool_info.flags = VK_DESCRIPTOR_POOL_CREATE_FREE_DESCRIPTOR_SET_BIT;
```

Comment:
This seems a bit risque if implementations do not have a good FREE_DESCRIPTOR_SET implementation. Would it be terrible if each root signature owns its own static pool/set?

Reply:
We can have a large number of root signatures and each descriptor pool is a memory object to track (for like 1-2 sampler descriptors most of the time); with this we'll still pack a large number of sets into a single pool even if the driver in question never reuses memory.

Reply:
I suppose. My only concern is if application allocates and frees root signatures all the time, and FREE_DESCRIPTOR_SET_BIT is not able to recycle memory internally, but I guess this is so unlikely to be a problem that it's irrelevant for our use case?

Reply:
I'd argue that if we ever find such a driver, we should push for that getting fixed instead.

Reply:
Fair enough.